### PR TITLE
SupportArticles: Make support article dialog a stand-alone module

### PR DESCRIPTION
**Pull request by spen**
This PR takes InlineHelp's support article view and turns it into a standalone component.
It also unravels some of the state related code and having redux take care of this.

The reason for doing this is to give the ability to launch a support article from anywhere in the app completely independent of `InlineHelp` - This should help us get the right information to folk who are looking for help or tips while using WordPress.com with a nicer experience :tada:

### Notes

I'd like to add documentation to this to explain things like how to use it from other places in the app etc. but I think it might be best to do this in a follow up PR.
There's also likely room for further trimming down or renaming things in InlineHelp, given the only dialog type that that module controls now is video.

### Testing

There should be no noticeable difference to how InlineHelp/support dialogs currently work.
To be sure of this, try opening inline help by clicking the blue `?` icon and finding an option that points to an article.
Hitting the 'read more' button that these options give should open a dialog containing one of our support articles as expected.
Try a few different articles, and try some of the other options (a video, guided tour and article can all be found at http://calypso.localhost:3000/media) and make sure those still work as expected too.

### CODE REVIEW

**Comment by nylen** [comment]
High-level question: why are both a postId and a url needed for the support article?

**Comment by spen** [comment]
> High-level question: why are both a postId and a url needed for the support article?

I'm fuzzy on the exact details but If I recall properly it's because we need a `postId` to make the API call required to get the post content but we ideally want a friendly link to allow folks to visit https://en.support.wordpress.com/that-article in situ.

I've done a bit of refactoring while shuffling files about that makes the `postUrl` param optional - omitting it just excludes the 'Visit Article' button from being rendered.

Ideally we'd be able to call one endpoint and receive this friendly url in that call, using that to render the 'visit article' button.

Another thought I've had is to utilize the data we have in `client/blocks/inline-help/contextual-help.js` to check if there is a pairing between a given `postId` and a matching `url` - that should be plausible but wouldn't be complete.

File: client/blocks/support-article-dialog/index.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	getDialogButtons() {
+		const { postUrl, translate } = this.props;
+		return [
+			postUrl ? (
+				<Button href={ postUrl } target="_blank" primary>
+					{ translate( 'Visit Article' ) } <Gridicon icon="external" size={ 12 } />
+				</Button>
+			) : (
+				<React.Fragment />
```
Comment (andfinally) [inline client/blocks/support-article-dialog/index.jsx]: Why do we have an empty fragment here instead of doing nothing?

**Comment by andfinally** [comment]
This looks great to me!

I did have one slight doubt about the naming - "support" in the name instead of "help" detaches this feature from inline help. On the other hand, I can see "support" will remind people of the support docs, which is probably what this'll mostly be used for.

I tested the different flavours of inline help on my local install in a few different sections - video, guided tour, article all still work fine.

File: client/blocks/support-article-dialog/index.jsx (hunk 1 above)
Comment (spen) [reply client/blocks/support-article-dialog/index.jsx]: If I'd have done `postUrl && ( <Button... /> )` we could have ended up with `false` in the array. Which, come to think about it wouldn't be a big issue I don't think. We could always just filter out  falseys as we return the array.

Comment (andfinally) [reply client/blocks/support-article-dialog/index.jsx]: Thanks! No worries, I was just curious.

**Comment by andfinally** [comment]
My pleasure Spen, I learned a lot!

**Pull request merged**